### PR TITLE
Fixes #28569 - consume jest from @theforeman

### PR DESCRIPTION
**Pull request by amirfefer**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,7 +22,7 @@
     "url": "http://projects.theforeman.org/projects/foreman-tasks/issues"
   },
   "dependencies": {
-    "@theforeman/vendor": "^3.3.2",
+    "@theforeman/vendor": "3.9.4-env-test.10",
```
Comment (amirfefer) [inline package.json]: this is a temporal version, **DO NOT** merge

File: package.json [hunk 2]
```diff
@@ -32,70 +32,24 @@
...
-    "babel-jest": "^24.9.0",
     "babel-loader": "^8.0.0",
     "coveralls": "^3.0.0",
-    "enzyme": "^3.4.0",
-    "enzyme-adapter-react-16": "^1.4.0",
-    "enzyme-to-json": "^3.2.1",
     "eslint": "^4.10.0",
     "eslint-import-resolver-babel-module": "^5.1.0",
     "eslint-plugin-patternfly-react": "0.2.0",
     "identity-obj-proxy": "^3.0.0",
```
Comment (sharvit) [inline package.json]: I believe this can be removed as well

File: package.json [hunk 3]
```diff
@@ -32,70 +32,24 @@
...
-    "@theforeman/env": "^3.3.2",
-    "@theforeman/builder": "^3.3.2",
-    "@theforeman/vendor-dev": "^3.3.2",
+    "@theforeman/test": "3.9.4-test-utils.0",
+    "@theforeman/builder": "3.9.4-test-utils.0",
+    "@theforeman/vendor-dev": "3.9.4-test-utils.0",
     "babel-eslint": "^10.0.3",
-    "babel-jest": "^24.9.0",
     "babel-loader": "^8.0.0",
     "coveralls": "^3.0.0",
```
Comment (sharvit) [inline package.json]: Can we remove `coveralls` as well?

File: webpack/ForemanTasks/Components/TaskDetails/__tests__/TaskDetailsActions.test.js [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import { testActionSnapshotWithFixtures } from 'react-redux-test-utils';
+import { testActionSnapshotWithFixtures } from '@theforeman/test';
```
Comment (sharvit) [inline webpack/ForemanTasks/Components/TaskDetails/__tests__/TaskDetailsActions.test.js]: IMO let's not migrate it yet, we can do it later and move the whole logic from `react-redux-test-utils` to `@theforeman/test`

File: webpack/ForemanTasks/Components/TasksDashboard/Components/TasksLabelsRow/TasksLabelsRow.test.js [hunk 5]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
-import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';
-import { shallow } from 'enzyme';
+import { testComponentSnapshotsWithFixtures } from '@theforeman/test';
+import { shallow } from '@theforeman/test';
```
Comment (sharvit) [inline webpack/ForemanTasks/Components/TasksDashboard/Components/TasksLabelsRow/TasksLabelsRow.test.js]: Can be imported in 1 line. Shouldn't eslint warning about it?

File: webpack/ForemanTasks/Components/common/ActionButtons/CancelButton.test.js [hunk 6]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
-import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';
-import { shallow } from 'enzyme';
+import { testComponentSnapshotsWithFixtures } from '@theforeman/test';
+import { shallow } from '@theforeman/test';
```
Comment (sharvit) [inline webpack/ForemanTasks/Components/common/ActionButtons/CancelButton.test.js]: Can be imported in 1 line

File: webpack/ForemanTasks/Routes/ForemanTasksRoutes.test.js [hunk 7]
```diff
@@ -1,6 +1,5 @@
 import React from 'react';
-import { shallow } from 'enzyme';
-import toJson from 'enzyme-to-json';
+import toJson, { shallow } from '@theforeman/test';
```
Comment (sharvit) [inline webpack/ForemanTasks/Routes/ForemanTasksRoutes.test.js]: I see it odd that the default import is the `toJson`, would expect to do `import { shallow, enzymeToJson } from '@theforeman/test';`

File: webpack/test_setup.js [hunk 8]
```diff
@@ -1,7 +1,3 @@
-import 'core-js/shim';
-import 'regenerator-runtime/runtime';
-
-import { configure } from 'enzyme';
-import Adapter from 'enzyme-adapter-react-16';
-
-configure({ adapter: new Adapter() });
+/* This is for extending test setup from @theforeman
```
Comment (sharvit) [inline webpack/test_setup.js]: Can't we remove this file and make it optional?

File: package.json [hunk 9]
```diff
@@ -22,7 +22,7 @@
     "url": "http://projects.theforeman.org/projects/foreman-tasks/issues"
   },
   "dependencies": {
-    "@theforeman/vendor": "^3.3.2",
+    "@theforeman/vendor": "3.9.4-test-utils.0",
```
Comment (sharvit) [inline package.json]: Should be changed to `3.10.0` once deployed.

File: webpack/ForemanTasks/Components/TaskDetails/__tests__/TaskDetailsActions.test.js (hunk 4 above)
Comment (amirfefer) [reply webpack/ForemanTasks/Components/TaskDetails/__tests__/TaskDetailsActions.test.js]: As we discussed offline, IMO `react-redux-test-utils` has a justification to stay standalone for open source sharing

File: webpack/test_setup.js (hunk 8 above)
Comment (amirfefer) [reply webpack/test_setup.js]: already done :)

**Comment by amirfefer** [comment]
Thanks @sharvit 
 rebased  :+1:

**Comment by adamruzicka** [comment]
Thank you @sharvit & @amirfefer !

File: package.json [hunk 10]
```diff
@@ -22,75 +22,26 @@
...
+    "@theforeman/vendor": "^4.0.2",
     "c3": "^0.4.11",
     "humanize-duration": "^3.20.1",
     "react-intl": "^2.8.0"
   },
   "devDependencies": {
     "@babel/core": "^7.7.0",
-    "@theforeman/env": "^3.11.0",
-    "@theforeman/builder": "^3.11.0",
-    "@theforeman/stories": "^3.11.0",
```
Comment (sharvit) [inline package.json]: It shouldn't remove the `@theforeman/stories` package :/

**Pull request merged**